frontend/public/components/utils/resource-log.jsx

```diff
@@ -155,17 +164,15 @@ export class ResourceLog extends SafetyFirst {
   // Toggle streaming/paused status
   _toggleStreaming() {
     const newStatus = this.state.status === 'streaming' ? 'paused' : 'streaming';
-    this.setState({
-      status: newStatus
-    });
+    this._updateStatus(newStatus);
   }
 
   // Updates log status
   _updateStatus(newStatus) {
     let newState = {status: newStatus};
 
     // Reset linesBehind when transitioning out of paused state
-    if (this.state.status === 'paused' && newStatus !== this.state.status) {
+    if (this.state.status !== 'streaming' && newStatus === 'streaming') {
```

Comment:
Might want to `Enum` the `status` values since you have numerous hard-coded `"streaming"` strings that could potentially typo.

Reply:
Yeah, didn't think about enum, this would be a good use case

Reply:
Can't use enum is jsx :(

Reply:
Ha, yeah. Didn't think about that. Punt till I update this component to tsx at some point. Wish I had done that up front :(

Reply:
or just use a `const streaming = 'streaming';

or 

```JavaScript 
const statuses = {
  streaming: 'streaming'
}
```
Either way, I think you get a sufficient improvement even if `Enum` isn't available yet.